### DISCUSSION TITLE
Pagespeed on Vercel and netlify

### DISCUSSION THREAD

**Discussion by Onibenjo** [vercel/vercel#4845] [Help] [answered]
I have my nextjs site hosted on vercel but running lighthouse on it seems to always produce a poor performance score than when lighthouse is ran on project hosted on netlify with about a 20 score difference.

Is there something I'm doing wrong? I really love vercel and don't want to move

**Comment by mcsdevv** [comment]
Hey @Onibenjo, thanks for letting us know, that's very odd, we nearly always find it to be the other way around.

Can you provide two identical deployments, one on each platform, along with the source repository, so we can investigate?

**Reply by Onibenjo** [reply]
@mcsdevv  Do you mean the build logs?

**Reply by mcsdevv** [reply]
Hello @Onibenjo, I mean two deployments, so we can compare side by side.

**Comment by Onibenjo** [comment]
Vercel - [Medtech](https://medtech.africa)

Netlify - [https://elegant-hodgkin-b72112.netlify.app/](https://elegant-hodgkin-b72112.netlify.app/)


@mcsdevv

**Reply by mcsdevv** [reply]
Hello @Onibenjo, these look like different sites to me, have you got two deployments with the same source code?

**Reply by Onibenjo** [reply]
@mcsdevv  link updated

**Reply by Onibenjo** [reply]
Metrics are way lower now because changes are being made to the site. Initially, I thought it was because netlify was using `next export` to statically export the files but that didn't seem to be the case

**Reply by markozxuu** [reply]
## Housed in Vercel
<img width="993" alt="Screen Shot 2020-07-16 at 10 48 41" src="https://user-images.githubusercontent.com/16585386/87693369-7f95f080-c752-11ea-97ec-c4bcdba8dbf4.png">

## Housed in Netlify
<img width="981" alt="Screen Shot 2020-07-16 at 10 49 28" src="https://user-images.githubusercontent.com/16585386/87693443-9a686500-c752-11ea-9110-052c94caef26.png">

## Conclusion

I am sure it is not the same code base that is deployed in Netlify 🤔

**Reply by markozxuu** [reply]
lighthouse's performance metrics are related to code optimization issues. It is very possible that you have 2 code bases that differ

**Reply by Onibenjo** [reply]
they are both connected to the same master branch

**Comment by styfle** [comment, marked as answer]
Thanks for sharing!

A quick comparison from my current location shows Vercel is consistently faster than Netlify.

Here's my chrome dev console with "Disable cache" checked. Notice the "Time" column and "Finish" at the bottom.

## Vercel
![Vercel](https://user-images.githubusercontent.com/229881/87710585-ee377600-c773-11ea-8fbb-235b7ff253ad.png)

## Netlify
![Netlify](https://user-images.githubusercontent.com/229881/87710691-0f986200-c774-11ea-9163-3ab71c2c9b3c.png)

-----

Which region are you closest to and what is your latency? 
https://vercel.com/edge-network

**Reply by styfle** [reply]
Oh second look, it seems that Google Analytics is the culprit. It's only running for the domain https://medtech.africa which is why you see the discrepancy.

**Reply by styfle** [reply]
Also this `setTimeout()` probably helps too

![image](https://user-images.githubusercontent.com/229881/87715162-eb8c4f00-c77a-11ea-864e-4d4b65364a18.png)

You'll need to make sure the code is the same or else you'll get different results 😄

**Reply by Onibenjo** [reply]
I am closest to Paris, France. CDG1
Latency: 140-198ms

**Reply by Onibenjo** [reply]
and the reason they had different code bases was that I was testing. So as at the time u tested it, I was probably striping things trying to get better lighthouse score

**Comment by empz** [comment]
I can vouch for syfle. I get consistent faster performance with the exact same Next.js site deployed on Vercel than on Netlify.

Vercel:
![image](https://user-images.githubusercontent.com/412489/94992495-a9f58e00-058a-11eb-8639-61b9f4c44089.png)

Netlify:
![image](https://user-images.githubusercontent.com/412489/94992508-b679e680-058a-11eb-8a05-e1047da00692.png)

Tests from Spain.
